src/lib/api/helper/interceptors.ts

```diff
@@ -111,22 +118,39 @@ export function responseInterceptor() {
         return response; // IF 2XX then return response.data only
     }
 }
-
-export function responseErrorInterceptor() {
+export function responseErrorInterceptor(axiosInstance: AxiosInstance) {
     return error => {
         // Request made and server responded
         if (error.response) {
             Debug(`Error Response  :  ${JSON.stringify(error.response.data)}`);
             throw new CommandError(`${error.response.data.message}`, ErrorCodes.API_ERROR.code);
         } else if (error.request) {
             // The request was made but no error.response was received
-            throw new Error(
-                'Not received response from the server, possibly some network issue, please retry!!'
-            );
+            
+            const originalRequest = error.config;
+            // Ask: Can I add custom header for this count?
+            // Check if it's network related error
+            // If yes, then add counter inside header
+            // If count is less than max retry limit, then resend request
+            if (["ECONNABORTED", "EPIPE", "ENOTFOUND", "ETIMEDOUT", "ECONNRESET"].includes(error.code)) {
```

Comment:
Can't we use an inbuilt package like here  https://www.npmjs.com/package/axios-retry?
@manishprajapati52

Reply:
As we discussed, please confirm, whether we want to use third party package or our own implementation?

Reply:
Please go-ahead with library @manishprajapati52